src/presenter/point-presenter.js

```diff
@@ -0,0 +1,112 @@
+import { render, replace } from '../framework/render.js';
+import EventsView from '../view/event-view.js';
+import EventEdit from '../view/event-edit.js';
+
+const PointMode = {
+  VIEW: 'view',
+  EDIT: 'edit',
+};
+
+export default class PointPresenter{
+
+  #eventsListView = null;
+  #point = null;
+  #offer = null;
+  #destination = null;
+  #points = null;
+
+  #pointComponent = null;
+  #pointEditComponent = null;
+  #openEditModeHandler = null;
+
+  #mode = null;
+  #modeFavorit = null;
+  #openFavoritModeHandler = null;
+  #changeFavorite = null;
+
+  constructor({eventsListView, point, offers, destinations}){
+    this.#eventsListView = eventsListView;
+    this.#point = point;
+    this.#offer = offers;
+    this.#destination = destinations;
+    this.#mode = PointMode.VIEW;
+    this.#modeFavorit = this.#point.isFavorite;
+  }
+
+  #renderPoint = (point) => {
```

Comment:
Добавил бы вспомогательный метод, который либо отрисовывает, либо заменяет уже отрисованный вариант
```suggestion
 #renderOrReplace(view) {
    const currentView = this.#mode === PointMode.VIEW ? this.#pointViewComponent : this.#pointEditComponent;
    if (currentView) {
      console.log('replace', currentView, view);
      replace(view, currentView);
    } else {
      console.log('render fresh', currentView, view);
      render(view, this.#eventsListView.element);
    }
  }
```